src/router-hooks.tsx

```diff
@@ -99,14 +110,7 @@ export const RouterProvider = ({ children }: { children: ReactNode }) => {
   }, [setState]);
   const navigate = useCallback(
     (newState: RouterState) => {
-      const query = Object.entries(newState)
-        .filter(([_, v]) => !!v)
-        .map(([k, v]) => {
-          return `${encodeURIComponent(k)}=${encodeURIComponent(
-            serializeValue(v)
-          )}`;
-        })
-        .join("&");
+      const query = getQueryString(newState);
       const url =
         window.location.toString().split("?")[0] + (query ? "?" + query : "");
```

Comment:
Is this bit common between the new usage and old? Perhaps a `toUrl(state: RouterState)` API would be better and fit both usecases?

Reply:
[`1fe7d6e`](https://github.com/microbit-foundation/python-editor-next/pull/477/commits/1fe7d6e76710879f26bd60a083836514a42eb880)